ehrapy/preprocessing/encoding/_encode.py

```diff
@@ -844,6 +844,13 @@ def _add_categoricals_to_obs(ann_data: AnnData, categorical_names: list[str]) ->
             continue
         elif var_name in categorical_names:
             ann_data.obs[var_name] = ann_data.X[::, idx : idx + 1]
+            # note: this will count binary columns (0 and 1 only) as well
+            # needed for writing to .h5ad files
+            if set(pd.unique(ann_data.obs[var_name])).issubset({False,True,np.NaN}):
```

Comment:
@Zethson: Implementation detail but: Due to pythons internal repr of bools and 0 and 1, this will also be true for any binary columns (with values 0,1 and maybe missing values). I wasn't able to find a way to only get True and False, since True==1 and False==0 is always valid in Python. Can you? Can u see a scenario where this could lead to issues?

Reply:
Mhmm. I think that the difficulty is that there's no consensus among the datasets that we use what is true and what is false for 0 and 1. Some say 0 is true and some say 0 is false.

Is this an issue?

Reply:
> Some say 0 is true and some say 0 is false.

I mean this is something that will lead to problems anyways and is nothing we can change. Its the general consensus IMO that 0 is False and 1 is True, at least in coding.

Reply:
I don't think that we can make this assumption for crazy data collectors. Can we parameterize this? By default we assume what you described, but it can be reversed? Not really possible here since we cannot control this behavior, right?...

Reply:
Na I would not support this or parametrize it. I dont think this will leads to issues since were not replacing the values, they remain 0 and 1, its just a internal repr detail, the users does not have to care about. 
As long as the user knows what is 0 and what is 1 , its good IMO.

Reply:
All right. Fine with me.